File: 2681-put-marbles-in-bags/2681-put-marbles-in-bags.cpp
class Solution {
public:
    long long putMarbles(vector<int>& weights, int k) {
        vector<int> a;
        int n = weights.size();
        for(int i = 0; i<n - 1; i++) a.push_back(weights[i] + weights[i + 1]);
        sort(a.begin(), a.end());
        long long max = 0, min = 0;
        for(int i= 0; i<k-1; i++) min += a[i];
        for(int i = n-2, t = 0; t<k-1; t++, i--) max += a[i];
        return max - min;
    }
};

File: 2681-put-marbles-in-bags/2681-put-marbles-in-bags.py
class Solution:
    def putMarbles(self, weights: List[int], k: int) -> int:
        a = []
        n = len(weights)
        for i in range(0, n-1):
            a.append(weights[i] + weights[i + 1])
        a.sort()
        mx = 0
        mn = 0
        for i in range(0, k - 1):
            mn += a[i]
        idx = n - 2
        for i in range(0, k - 1):
            mx += a[idx]
            idx -= 1
        return mx - mn